Remotion/Data/DomainObjects.UnitTests/DataManagement/DataContainerStateTest.cs

```diff
@@ -167,5 +196,28 @@ public void To_String ()
       var state = new DomainObjectState.Builder().SetInvalid().SetDeleted().Value;
       Assert.That(state.ToString(), Is.EqualTo("DomainObjectState (Deleted, Invalid)"));
     }
+
+    private void CheckOnlySingleFlagIsSet (DataContainerState state)
+    {
+      int count = 0;
+      if (state.IsNew)
+        count++;
+      if (state.IsNewInHierarchy)
+        count++;
+      if (state.IsChanged)
+        count++;
+      if (state.IsDeleted)
+        count++;
+      if (state.IsDiscarded)
+        count++;
+      if (state.IsUnchanged)
+        count++;
+      if (state.IsPersistentDataChanged)
+        count++;
+      if (state.IsNonPersistentDataChanged)
+        count++;
+
+      Assert.That(count, Is.EqualTo(1));
```

Comment:
Some of the flag values are not covered? Is that intended?

Reply:
@a-ctor which flags aren't covered? I'm counting 8 flags declared on the enum and 8 flags covered.

Reply:
Ok I might have mistakenly counted properties in the wrong enum :D